mk/vrbrain_targets.mk

```diff
@@ -1,467 +1,264 @@
-
-$(VRBRAIN_ROOT)/Archives/vrbrain-v51.export:
-	$(v) $(VRBRAIN_MAKE_ARCHIVES)
-
-$(VRBRAIN_ROOT)/Archives/vrbrain-v51P.export:
-	$(v) $(VRBRAIN_MAKE_ARCHIVES)
-
-$(VRBRAIN_ROOT)/Archives/vrbrain-v51Pro.export:
-	$(v) $(VRBRAIN_MAKE_ARCHIVES)
-
-$(VRBRAIN_ROOT)/Archives/vrbrain-v51ProP.export:
-	$(v) $(VRBRAIN_MAKE_ARCHIVES)
-
-$(VRBRAIN_ROOT)/Archives/vrbrain-v52.export:
-	$(v) $(VRBRAIN_MAKE_ARCHIVES)
-
-$(VRBRAIN_ROOT)/Archives/vrbrain-v52P.export:
-	$(v) $(VRBRAIN_MAKE_ARCHIVES)
-
-$(VRBRAIN_ROOT)/Archives/vrbrain-v52Pro.export:
-	$(v) $(VRBRAIN_MAKE_ARCHIVES)
-
-$(VRBRAIN_ROOT)/Archives/vrbrain-v52ProP.export:
-	$(v) $(VRBRAIN_MAKE_ARCHIVES)
-
-$(VRBRAIN_ROOT)/Archives/vrubrain-v51.export:
-	$(v) $(VRBRAIN_MAKE_ARCHIVES)
-
-$(VRBRAIN_ROOT)/Archives/vrubrain-v51P.export:
-	$(v) $(VRBRAIN_MAKE_ARCHIVES)
-
-$(VRBRAIN_ROOT)/Archives/vrubrain-v52.export:
-	$(v) $(VRBRAIN_MAKE_ARCHIVES)
-
-$(VRBRAIN_ROOT)/Archives/vrubrain-v52P.export:
-	$(v) $(VRBRAIN_MAKE_ARCHIVES)
-
-$(VRBRAIN_ROOT)/Archives/vrhero-v10.export:
-	$(v) $(VRBRAIN_MAKE_ARCHIVES)
-
-vrbrain-archives:
-	$(v) $(VRBRAIN_MAKE_ARCHIVES)
-
-else
-
-vrbrain_nx:
-	$(error ERROR: You need to add VRBRAIN_ROOT to your config.mk)
-
-vrbrain-clean: vrbrain
-
-vrbrain-upload: vrbrain
-
-endif
+# VRBRAIN build is via external build system
+
+ifneq ($(VRBRAIN_ROOT),)
+$(error VRBRAIN_ROOT found in config.mk - Please see http://dev.ardupilot.org/wiki/git-submodules/)
+endif
+
+ifneq ($(NUTTX_SRC),)
+$(error NUTTX_SRC found in config.mk - Please see http://dev.ardupilot.org/wiki/git-submodules/)
+endif
+
+ifneq ($(UAVCAN_DIR),)
+$(error UAVCAN_DIR found in config.mk - Please see http://dev.ardupilot.org/wiki/git-submodules/)
+endif
+
+# these can be overridden in developer.mk
+VRBRAINFIRMWARE_DIRECTORY ?= $(SKETCHBOOK)/modules/PX4Firmware
+VRBRAINNUTTX_DIRECTORY ?= $(SKETCHBOOK)/modules/PX4NuttX
+UAVCAN_DIRECTORY ?= $(SKETCHBOOK)/modules/uavcan
+
+VRBRAIN_ROOT := $(shell cd $(VRBRAINFIRMWARE_DIRECTORY) && pwd)
+NUTTX_ROOT := $(shell cd $(VRBRAINNUTTX_DIRECTORY) && pwd)
+NUTTX_SRC := $(NUTTX_ROOT)/nuttx/
+UAVCAN_DIR=$(shell cd $(UAVCAN_DIRECTORY) && pwd)/
+
+# warn if user has old PX4Firmware or PX4NuttX trees
+ifneq ($(wildcard $(SKETCHBOOK)/../PX4Firmware),)
+$(warning *** You have an old PX4Firmware tree - see http://dev.ardupilot.com/wiki/git-submodules/)
+endif
+ifneq ($(wildcard $(SKETCHBOOK)/../PX4NuttX),)
+$(warning *** You have an old PX4NuttX tree - see http://dev.ardupilot.com/wiki/git-submodules/)
+endif
+ifneq ($(wildcard $(SKETCHBOOK)/../uavcan),)
+$(warning *** You have an old uavcan tree - see http://dev.ardupilot.com/wiki/git-submodules/)
+endif
+
+NUTTX_GIT_VERSION ?= $(shell cd $(NUTTX_SRC) && git rev-parse HEAD | cut -c1-8)
+PX4_GIT_VERSION   ?= $(shell cd $(VRBRAIN_ROOT) && git rev-parse HEAD | cut -c1-8)
+
+EXTRAFLAGS += -DNUTTX_GIT_VERSION="\"$(NUTTX_GIT_VERSION)\""
+EXTRAFLAGS += -DPX4_GIT_VERSION="\"$(PX4_GIT_VERSION)\""
+EXTRAFLAGS += -DUAVCAN=1

```

Comment:
I'm not sure where this define is used but am I mistaken that VRBRAIN doesn't have UAVCAN?